lib/email/utils/helpers.js

```diff
@@ -28,7 +30,7 @@ function logFlowEventFromMessage(log, message, type) {
   const elapsedTime = currentTime - flowBeginTime
 
   if (flowId && flowBeginTime && (elapsedTime > 0) && type && templateName) {
-    const eventName = `email.${templateName}.${type}`
+    const eventName = `email.${templateName}.${type}.${emailDomain}`
```

Comment:
> To be sure, you are proposing something like this

Broadly, yes.  Although you might not even need to include the flow_* properties in there, since not all email-related events will have them.